client/dashboard/app/router.tsx

```diff
@@ -180,6 +180,17 @@ const siteSettingsWordPressRoute = createRoute( {
 	)
 );
 
+const siteSettingsStaticFile404Route = createRoute( {
```

Comment:
I believe we also need to fetch the query in the loader... see https://github.com/Automattic/wp-calypso/pull/103560#discussion_r2102220675